src/Vectorface/SnappyRouter/Request/HttpRequest.php

```diff
@@ -35,14 +42,18 @@ class HttpRequest extends AbstractRequest implements HttpRequestInterface
      * @param string $controller The controller being requested.
      * @param string $action The action being invoked.
      * @param string $verb The HTTP verb used in the request.
+     * @param mixed $stream A stream or a string describing a stream location.
      */
-    public function __construct($controller, $action, $verb)
+    public function __construct($controller, $action, $verb, $stream)
     {
         parent::__construct($controller, $action);
         $this->setVerb($verb);
+        $this->setStream($stream);
+
         $this->input = array(
             self::INPUT_METHOD_QUERY  => array(),
-            self::INPUT_METHOD_POST   => array()
+            self::INPUT_METHOD_POST   => array(),
+            self::INPUT_METHOD_STREAM => ''
```

Comment:
A few concerns about this:
 - Naming it input method stream is a little awkward. (This may not be a valid concern, as input method (get|post) is also a little awkward to me.)
 - The value should be null or false so we can tell the difference between an empty body (Content-Length: 0) and no body whatsoever (e.g. GET request).